e2etests/pkg/k8s/nodes.go

```diff
@@ -6,16 +6,40 @@ import (
 	"context"
 	"fmt"
 
-	v1 "k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	applyconfigurationscorev1 "k8s.io/client-go/applyconfigurations/core/v1"
 	clientset "k8s.io/client-go/kubernetes"
 )
 
-func GetNodes(cs clientset.Interface) ([]v1.Node, error) {
+const (
+	nodeLabelerFieldManagerName = "openpe-e2e-node-labeler"
+)
+
+func GetNodes(cs clientset.Interface) ([]corev1.Node, error) {
 	nodes, err := cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
 	if err != nil {
 		return nil, fmt.Errorf("failed to list nodes: %w", err)
 	}
 
 	return nodes.Items, nil
 }
+
+func LabelNodes(cs clientset.Interface, labels map[string]string, nodes ...corev1.Node) error {
+	for _, node := range nodes {
+		_, err := cs.CoreV1().Nodes().Apply(context.Background(), applyconfigurationscorev1.
+			Node(node.Name).
+			WithLabels(labels), metav1.ApplyOptions{
+			FieldManager: nodeLabelerFieldManagerName,
+		})
```

Comment:
IMHO this is hard to parse. Please ensure that this is properly nested, something like:
```golang
		_, err := cs.CoreV1().Nodes().Apply(
			context.Background(),
			applyconfigurationscorev1.
				Node(node.Name).
				WithLabels(labels),
			metav1.ApplyOptions{
				FieldManager: nodeLabelerFieldManagerName,
			},
		)
```